kernel/src/device/fb.rs

```diff
@@ -325,6 +287,109 @@ impl FbHandle {
         current_userspace!().write_val(arg, &screen_info)?;
         Ok(0)
     }
+
+    /// Handles the [`IoctlCmd::GETCMAP`] ioctl command.
+    /// Takes a [`aster_framebuffer::FbCmap`] structure specifying the range as input and outputs the same structure filled with color palette data.
+    fn handle_get_cmap(&self, arg: usize) -> Result<i32> {
+        let cmap_user: FbCmapUser = current_userspace!().read_val(arg)?;
+
+        if cmap_user.len == 0 {
+            return Ok(0);
+        }
+
+        let cmap = self.framebuffer.cmap().lock();
+
+        // Check bounds
+        let start = cmap_user.start as usize;
+        let len = cmap_user.len as usize;
+
+        if start >= cmap.red.len() || start + len > cmap.red.len() {
```

Comment:
```suggestion
        if start >= cmap.red.len() || len > cmap.red.len() - start {
```

`start + len` can overflow, which will make the kernel panic in debug builds.